libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -139,7 +141,11 @@ string IRGenerator::generateFunction(FunctionDefinition const& _function)
 		t("functionName", functionName);
 		string params;
 		for (auto const& varDecl: _function.parameters())
-			params += (params.empty() ? "" : ", ") + m_context.addLocalVariable(*varDecl).commaSeparatedList();
+			params += (params.empty() ? "" : ", ") +
+				(m_context.isLocalVariable(*varDecl) ?
+					IRVariable{*varDecl}.commaSeparatedList() :
+					m_context.addLocalVariable(*varDecl).commaSeparatedList()
+				);
```

Comment:
I needed to add this, as I add param as a local variable in implicit constructor as it is needed for evaluation of base constructor parameters
Other option is to somehow clear local variables between functions
@chriseth what do you think?

Reply:
I do not fully understand this...

Reply:
In generateImplicitConstructors it does add this variable, than in user defined if I add it again it complains
```
if (contract->constructor())
				for (ASTPointer<VariableDeclaration> const& varDecl: contract->constructor()->parameters())
					params += (params.empty() ? "" : ", ") + m_context.addLocalVariable(*varDecl).commaSeparatedList();
```

Reply:
Ah I see - so this is the downside of not inlining the body of the user-defined constructor?

Reply:
Maybe we should then actually rather inline it. Let's choose the simpler solution for now (I hope it is simpler) and see if we have to switch back once we implement modifiers.